src/controllers/simplified-account/settings/card-payments/apple-pay.controller.js

```diff
@@ -0,0 +1,31 @@
+const { response } = require('@utils/response')
+const { formatSimplifiedAccountPathsFor } = require('@utils/simplified-account/format')
+const { ConnectorClient } = require('@services/clients/connector.client')
+const { onOrOffToBool } = require('@utils/on-or-off')
+const paths = require('@root/paths')
+
+const connector = new ConnectorClient(process.env.CONNECTOR_URL)
+
+function get (req, res) {
+  const url = req.originalUrl.split('?')[0]
+  const applePay = req.account?.allowApplePay
+  response(req, res, 'simplified-account/settings/card-payments/apple-pay', {
+    formAction: url,
+    currentState: applePay ? 'on' : 'off',
+    backLink: formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.cardPayments.index, req.service.externalId, req.account.type)
+  })
+}
+
+async function post (req, res) {
+  const userPreference = onOrOffToBool(req.body.applePay)
+  const serviceExternalId = req.service.externalId
+  const gatewayAccountId = req.account.id
+  await connector.toggleApplePay(gatewayAccountId, userPreference)
```

Comment:
something like 
```suggestion
toggleApplePay(gatewayAccountId, userPreference)
  .then(() => {
    req.flash('messages', { state: 'success', icon: '&check;', heading: `Apple Pay successfully ${userPreference ? 'enabled' : 'disabled'}` })
    res.redirect(formatSimplifiedAccountPathsFor(paths.simplifiedAccount.settings.cardPayments.index, serviceExternalId, req.account.type))
  })
  .catch((err) => {
    // handle specific error
    next(err) // default error handling
  })
```

Reply:
Sorry, forgot to comment on this one after going via a service. Was that all this suggestion was showing? As far as I can tell the explicit promise handling is not doing anything that the await isn't doing implicitly.

Reply:
just a suggestion, doesn't need to be changed